### DISCUSSION TITLE
Best way to pass data to child modal (e.g. multi-step form)

### DISCUSSION THREAD

**Discussion by bluec** [wire-elements/modal#178] [General]
Hi, apologies is this has been addressed elsewhere I did have a good look.

What's the best way to pass data to a child modal, for example in a multi-step form?

Currently I am validating the data in step 1 modal, then passing it to the mount method of step 2 modal like so:

```php
class Step1Modal extends ModalComponent
{
    public $name;
    public $email;
    public $telephone;

   [...]

    public function submit()
    {
        $this->validate();

        $this->emit(
            'openModal',
            'modal.step2-modal',
            [
                'name' => $this->name,
                'email' => $this->email,
                'telephone' => $this->telephone
            ]
        );
    }
}

class Step2Modal extends ModalComponent
{
    public $name;
    public $email;
    public $telephone;

   [...]

    public function mount($name, $email, $telephone)
    {
        $this->name = $name;
        $this->email = $email;
        $this->telephone = $telephone;
    }

```

I then have to validate the step 1 data again in step 2 modal when that is submitted otherwise the validation could be by-passed.

Is there a better way to handle this?

**Comment by PhiloNL** [comment]
I would recommend keeping all the steps in the same model rather than having separate modals. As you mentioned, having separate modals will require you to validate everything again in each component.

**Reply by bluec** [reply]
Ok thanks for the reply. I wondered if I was missing something. Not ideal as one thing that drew me to this package was the child modal flow. I guess I can re-validate in each step but it feels a bit messy. I will keep pondering it! Many thanks
